genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncher.java

```diff
@@ -0,0 +1,226 @@
+            final long end = System.nanoTime();
+            this.registry.timer(LAUNCH_TIMER, tags).record(end - start, TimeUnit.NANOSECONDS);
+            this.healthIndicatorCache.put(jobId, titusJobId);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Health health() {
+        return Health.up()
+            .withDetails(this.healthIndicatorCache.asMap())
+            .build();
+    }
+
+    private TitusBatchJobRequest createJobRequest(final ResolvedJob resolvedJob) {
+
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        // Map placeholders in entry point template to their values
+        final Map<String, String> placeholdersMap = ImmutableMap.of(
+            TitusAgentLauncherProperties.JOB_ID_PLACEHOLDER,
+            jobId,
+            TitusAgentLauncherProperties.SERVER_HOST_PLACEHOLDER,
+            this.titusAgentLauncherProperties.getGenieServerHost(),
+            TitusAgentLauncherProperties.SERVER_PORT_PLACEHOLDER,
+            String.valueOf(this.titusAgentLauncherProperties.getGenieServerPort())
+        );
+
+        // Substitute all placeholders with their values
+        final List<String> entryPoint =
+            this.titusAgentLauncherProperties.getEntryPointTemplate()
+                .stream()
+                .map(s -> placeholdersMap.getOrDefault(s, s))
+                .collect(Collectors.toList());
+
+        return new TitusBatchJobRequest(
+            new TitusBatchJobRequest.Owner(this.titusAgentLauncherProperties.getOwnerEmail()),
+            this.titusAgentLauncherProperties.getApplicationName(),
+            this.titusAgentLauncherProperties.getCapacityGroup(),
+            ImmutableMap.of(
+                GENIE_USER_ATTR, resolvedJob.getJobMetadata().getUser(),
+                GENIE_SOURCE_HOST_ATTR, this.genieHostInfo.getHostname(),
+                GENIE_ENDPOINT_ATTR, this.titusAgentLauncherProperties.getGenieServerHost(),
+                GENIE_JOB_ID_ATTR, jobId
+            ),
+            new TitusBatchJobRequest.Container(
+                new TitusBatchJobRequest.Resources(
+                    resolvedJob.getJobEnvironment().getCpu(),
+                    0,
+                    resolvedJob.getJobEnvironment().getMemory(),
+                    this.titusAgentLauncherProperties.getDiskSize().toMegabytes(),
+                    this.titusAgentLauncherProperties.getNetworkBandwidth().toMegabytes() * 8 //MB to Mb
+                ),
+                new TitusBatchJobRequest.SecurityProfile(
+                    this.titusAgentLauncherProperties.getSecurityAttributes(),
+                    this.titusAgentLauncherProperties.getSecurityGroups(),
+                    this.titusAgentLauncherProperties.getIAmRole()
+                ),
+                new TitusBatchJobRequest.Image(
+                    this.titusAgentLauncherProperties.getImageName(),
+                    this.titusAgentLauncherProperties.getImageTag()
```

Comment:
It would be nice if a bunch of these properties were dynamic. In particular these two as if we have a bad image deployment using a fast property to pin it back to previous version quickly would be useful

Reply:
It may even go further than that.. for example you may want to use different images for different job types.

For now, I'm making the smallest step from immutable host to immutable property image.
But i can see this evolving and this property going away completely.